RemoteContainers.md

```diff
@@ -0,0 +1,185 @@
+
+## Prerequisites
+
+There might be two options of using Visual Studio Code remote containers with m365pnp/spfx image:
+
+- Using locally installed Docker, on your desktop/laptop
+- Running containers in cloud using GitHub Codespaces (requires a paid GitHub plan)
+
+When you run in a remote container in GitHub Codespaces, you do NOT need to install any of the following on your laptop/desktop:
+
+- Node.js
+- yeoman
+- gulp
+- Docker
+- git
+
+What you will need locally is only Visual Studio!
+
+### Locally Installed Docker
+
+For using that option you need to have locally installed the following software:
+
+- Visual Studio Code
+- Visual Studio Code extansion, Remote - Containers https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers
+- Docker Engine
+- SPFx solution code, scaffolded in the root directory
+
+### GitHub Codespaces
+
+This option requires the following:
+
+- Visual Studio Code
+- Visual Studio Code extansion, Remote - Containers, https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers
+- Git repository in a GitHub organization with enabled Codespaces, https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/enabling-github-codespaces-for-your-organization
+- SPFx solution code, scaffolded in the root directory
+
+## Defining Remote Containers for the Project
+
+1. In the root folder of the source code, create the folder called `.devcontainer`.
+
+2. In this folder, create the `devcontainer.json` file with the following contents:
+
+```json
+// For more information on how to run this SPFx project in a VS Code Remote Container, please visit https://aka.ms/spfx-devcontainer
+{
+	"name": "SPFx 1.15.0",
+	"image": "m365pnp/spfx:1.15.0",
+	// Set *default* container specific settings.json values on container create.
+	"settings": {},
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"editorconfig.editorconfig",
+		"dbaeumer.vscode-eslint"
+	],
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	"forwardPorts": [
+		4321,
+		35729
+	],
+	"portsAttributes": {
+		"4321": {
+			"label": "Manifest",
+			"requireLocalPort": true
+		},
+		"35729": {
+			"label": "LiveReload",
+			"requireLocalPort": true
+		}
+	},
+	"postCreateCommand": "bash .devcontainer/spfx-startup.sh",
+	"remoteUser": "node"
+}
+```
+
+3. In the same folder create the `spfx-startup.sh` file with the following contents:
+
+```sh
+echo
+echo -e "\e[1;94mInstalling Node dependencies\e[0m"
+npm install
+
+## commands to create dev certificate and copy it to the root folder of the project
+echo
+echo -e "\e[1;94mGenerating dev certificate\e[0m"
+gulp trust-dev-cert
+```
+
+This is the structure that you should have:
+
+![](./docu-pictures/vs-files.png)
```

Comment:
Let's rename the `docu-pictures` folder to `images` or `assets`

Reply:
done